tests/test_forms.py

```diff
@@ -27,6 +27,29 @@ def test_yearless_date_not_supplied(self):
         assert not form.is_valid()
         assert form.errors['yearless_date'] == [u'This field is required.']
 
+    def test_yearless_date_form_has_not_changed(self):
+        form = YearlessDateForm(initial={'yearless_date': YearlessDate(1, 9)},
+                                data={'yearless_date_0': '1', 'yearless_date_1': '9'})
+        assert not form.has_changed()
+
+    def test_yearless_date_form_has_changed_value_to_empty(self):
+        form = YearlessDateForm(initial={'yearless_date': YearlessDate(1, 9)},
+                                data={'yearless_date_0': '', 'yearless_date_1': ''})
+        assert form.has_changed()
+
+    def test_yearless_date_form_has_changed_value_to_new_value(self):
+        form = YearlessDateForm(initial={'yearless_date': YearlessDate(29, 10)},
+                                data={'yearless_date_0': '', 'yearless_date_1': ''})
```

Comment:
The data should be populated with a new date, no?

Reply:
The form is initialized with a date (October 29), but then the form data is updated to have an empty datefield (data object). This test checks to ensure that the form has detected the change, and that change would actually make it so the field is None.

Reply:
The test could be renamed to `test_yearless_date_form_has_changed_value_to_none`

Reply:
Wouldn't that make it the same as the previous test (`test_yearless_date_form_has_changed_value_to_empty`)? Probably better to have this test change one value to another, as the test name implies?

Reply:
Good catch, my bad. I'll push a fix.